### PR TITLE
APR-2793 APR-2794 implemented download files activity and workflow

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.Charities.Import/Startup.cs [hunk 1]
```diff
@@ -11,6 +11,8 @@ internal class Startup : FunctionsStartup
     {
         public override void Configure(IFunctionsHostBuilder builder)
         {
+            builder.Services.AddHttpClient();
```
Comment (martin-lush) [inline src/SFA.DAS.Charities.Import/Startup.cs:14]: Comment: Default HandlerLifeTime used. Are you sure 2 minutes is enough for your resource pool?

File: src/SFA.DAS.Charities.Import/Functions/ImportCharityCommissionData/Activities/ImportCharityCommissionDataActivity.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            _httpClient = httpClient;
+        }
+
+        [FunctionName(nameof(ImportCharityCommissionDataActivity))]
+        public async Task Run(
+            [ActivityTrigger] string fileName, 
+            [Blob("charity-files/{fileName}", FileAccess.Write)] Stream file,
+            ILogger log)
+        {
+            var charityFileDownloadUri = new Uri(Path.Combine(downloadUrl, fileName));
```
Comment (martin-lush) [inline src/SFA.DAS.Charities.Import/Functions/ImportCharityCommissionData/Activities/ImportCharityCommissionDataActivity.cs:31]: Normally you set the base address during Dependency injection setup and have it as either an named instance or specified Interface

File: src/SFA.DAS.Charities.Import/Functions/ImportCharityCommissionData/Activities/ImportCharityCommissionDataActivity.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            log.LogDebug("Downloading file: {charityFileDownloadUrl}", charityFileDownloadUri);
+
+            byte[] response;
+
+            try
+            {
+                response = await _httpClient.GetByteArrayAsync(charityFileDownloadUri);
+            }
+            catch (Exception ex)
```
Comment (martin-lush) [inline src/SFA.DAS.Charities.Import/Functions/ImportCharityCommissionData/Activities/ImportCharityCommissionDataActivity.cs:40]: Shouldn't be catching a generic Exception here. Also you're not checking for a successful response code

**Review by martin-lush — APPROVED** [review]

File: src/SFA.DAS.Charities.Import/Startup.cs (hunk 1 above)
Comment (DevenKShah) [reply src/SFA.DAS.Charities.Import/Startup.cs]: Configured to be 5

File: src/SFA.DAS.Charities.Import/Functions/ImportCharityCommissionData/Activities/ImportCharityCommissionDataActivity.cs (hunk 3 above)
Comment (DevenKShah) [reply src/SFA.DAS.Charities.Import/Functions/ImportCharityCommissionData/Activities/ImportCharityCommissionDataActivity.cs]: handling HttpRequestException

File: src/SFA.DAS.Charities.Import/Functions/ImportCharityCommissionData/Activities/ImportCharityCommissionDataActivity.cs (hunk 2 above)
Comment (DevenKShah) [reply src/SFA.DAS.Charities.Import/Functions/ImportCharityCommissionData/Activities/ImportCharityCommissionDataActivity.cs]: Fixed

**Pull request merged**